### PR TITLE
[SYCLomatic] Support migration of 6 mipmapped array APIs.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

**Review by ziranzha — APPROVED** [review]

File: clang/lib/DPCT/APINamesMemory.inc [hunk 1]
```diff
@@ -171,6 +171,41 @@ ASSIGNABLE_FACTORY(CONDITIONAL_FACTORY_ENTRY(
...
                       getSizeForMalloc(0, 1), QUEUESTR))))))
 
+CONDITIONAL_FACTORY_ENTRY(
+    UseExtBindlessImages,
+    ASSIGNABLE_FACTORY(DELETER_FACTORY_ENTRY("cudaFreeMipmappedArray", ARG(0))),
+    REMOVE_API_FACTORY_ENTRY_WITH_MSG(
+        "cudaFreeMipmappedArray",
+        "SYCL currently does not support mipmap image type. You "
+        "can migrate the code with bindless images by specifying "
+        "--use-experimental-features=bindless_images."))
```
Comment (zhimingwang36) [inline clang/lib/DPCT/APINamesMemory.inc]: for the string " SYCL currently .......suggest to ....", suggest to define a constant string, then can be used everywhere.

File: clang/runtime/dpct-rt/include/dpct/image.hpp [hunk 2]
```diff
@@ -471,48 +471,89 @@ class sampling_info {
   sycl::filtering_mode _filtering_mode = sycl::filtering_mode::nearest;
   sycl::coordinate_normalization_mode _coordinate_normalization_mode =
       sycl::coordinate_normalization_mode::unnormalized;
+  sycl::filtering_mode _mipmap_filtering = sycl::filtering_mode::nearest;
+  float _min_mipmap_level_clamp = 0.f;
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/dpct/image.hpp:478]: the new added 4 attributes is not so straightforward,  please add comment on it and it's get/set interface.

File: clang/lib/DPCT/APINamesMemory.inc (hunk 1 above)
Comment (tangjj11) [reply clang/lib/DPCT/APINamesMemory.inc]: fixed

File: clang/runtime/dpct-rt/include/dpct/image.hpp (hunk 2 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/image.hpp:478]: fixed

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**